src/Shared/GatewaySurchargeHandler.php

```diff
@@ -270,7 +270,9 @@ protected function orderAddFee($order, $amount, $surchargeName)
 
     protected function canProcessOrder()
     {
-        $orderId = !empty(filter_var(wp_unslash($_POST['orderId']), FILTER_SANITIZE_NUMBER_INT)) ? filter_var(wp_unslash($_POST['orderId']), FILTER_SANITIZE_NUMBER_INT) : false;
+        $orderId = !empty(filter_input(INPUT_POST, 'orderId', FILTER_SANITIZE_NUMBER_INT))
+                ? filter_input(INPUT_POST, 'orderId', FILTER_SANITIZE_NUMBER_INT)
```

Comment:
You should save the value to a variable first to

- avoid duplication
- make it more correct (theoretically a function may return different values)
- avoid psalm warnings